### PR TITLE
nvidia_x11.vulkan_beta: 450.56.11 -> 455.46.01

### PR DESCRIPTION
**Pull request by expipiplus1**
~Still need to test, hence WIP~ Seems to work well

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Can you rebase this branch on master to fix the eval error?

**Comment by expipiplus1** [comment]
Done, I also bumped it to 455.46.02, released yesterday.

**Comment by expipiplus1** [comment]
Is it possible to get this merged please?

**Comment by expipiplus1** [comment]
ping..

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104455` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages_hardkernel_4_14.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_hardkernel_latest.nvidia_x11_vulkan_beta</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>linuxPackages-libre.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages.nvidia_x11_vulkan_beta (linuxPackages_5_4.nvidia_x11_vulkan_beta)</li>
    <li>linuxPackages_4_14.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_4_19.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_4_4.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_4_9.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_5_8.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_5_9.nvidia_x11_vulkan_beta (linuxPackages_latest.nvidia_x11_vulkan_beta)</li>
    <li>linuxPackages_hardened.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_latest-libre.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_latest_hardened.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_latest_xen_dom0.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_latest_xen_dom0_hardened.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_testing_bcachefs.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_xen_dom0.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_xen_dom0_hardened.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_zen.nvidia_x11_vulkan_beta</li>
  </ul>
</details>

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 1]
```diff
@@ -33,14 +33,17 @@ rec {
...
+  vulkan_beta = generic rec {
+    version = "455.46.02";
+    persistencedVersion = "455.45.01";
+    settingsVersion = "455.45.01";
+    sha256_64bit = "05y4scg62mrhjnj2c8689m5hdyzqnx5p8vqvdqvv76zy970723l7";
+    settingsSha256 = "09v86y2c8xas9ql0bqr7vrjxx3if6javccwjzyly11dzffm02h7g";
+    persistencedSha256 = "13s4b73il0lq2hs81q03176n16mng737bfsp3bxnxgnrv3whrayz";
+    url =
+      let versionNoDots = lib.concatStrings (lib.splitString "." version);
+      in "https://developer.nvidia.com/vulkan-beta-${versionNoDots}-linux";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/nvidia-x11/default.nix:46]:
```suggestion
    url = "https://developer.nvidia.com/vulkan-beta-${lib.concatStrings (lib.splitString "." version)}-linux";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/nvidia-x11/default.nix (hunk 1 above)
Comment (expipiplus1) [reply pkgs/os-specific/linux/nvidia-x11/default.nix:46]: I don't agree that this proposed change makes the code more readable or maintainable.

Comment (SuperSandro2000) [reply pkgs/os-specific/linux/nvidia-x11/default.nix:46]: I did a quick ``rg concatStrings`` and most places do it like my proposed change.

Comment (expipiplus1) [reply pkgs/os-specific/linux/nvidia-x11/default.nix:46]: But it's nothing to do with `concatStrings` specifically. It's about making a computation clearer by giving a name to an intermediate stage.

I don't consider this important enough to warrant further argument, so will make the change if you say you insist.

Comment (SuperSandro2000) [reply pkgs/os-specific/linux/nvidia-x11/default.nix:46]: When I searched I saw a lot of cases like my proposed change. I am insisting on it.

Comment (expipiplus1) [reply pkgs/os-specific/linux/nvidia-x11/default.nix]: done

**Comment by expipiplus1** [comment]
Is it possible to merge this please?

**Comment by expipiplus1** [comment]
Thanks

**Pull request merged**